packages/manager/apps/web/client/app/hosting/cdn/order/hosting-cdn-order.routing.js

```diff
@@ -1,105 +1,251 @@
 import find from 'lodash/find';
 import get from 'lodash/get';
+import includes from 'lodash/includes';
+import assign from 'lodash/assign';
+
+import {
+  HOSTING_CDN_ORDER_CDN_VERSION_V1,
+  HOSTING_CDN_ORDER_CDN_VERSION_V2,
+} from './hosting-cdn-order.constant';
 
 export default /* @ngInject */ ($stateProvider) => {
-  $stateProvider.state('app.hosting.cdn.order', {
-    url: '/order',
-    component: 'hostingCdnOrder',
-    resolve: {
-      /* @ngInject */
-      autoPayWithPreferredPaymentMethod: (ovhPaymentMethod) =>
-        ovhPaymentMethod.hasDefaultPaymentMethod(),
+  const resolve = {
+    /* @ngInject */
+    autoPayWithPreferredPaymentMethod: (ovhPaymentMethod) =>
+      ovhPaymentMethod.hasDefaultPaymentMethod(),
+
+    catalogAddons: /* @ngInject */ (user, WucOrderCartService) =>
+      WucOrderCartService.getProductPublicCatalog(
+        user.ovhSubsidiary,
+        'webHosting',
+      ),
 
-      catalogAddon: /* @ngInject */ (
-        goBackWithError,
+    catalogAddon: /* @ngInject */ (
+      goBackWithError,
+      serviceOption,
+      user,
+      $translate,
+      HostingCdnOrderService,
+    ) =>
+      HostingCdnOrderService.getCatalogAddon(
+        user.ovhSubsidiary,
         serviceOption,
-        user,
-        $translate,
-        HostingCdnOrderService,
-      ) =>
-        HostingCdnOrderService.getCatalogAddon(
+      ).catch((error) => goBackWithError(get(error, 'data.message', error))),
+
+    goBack: /* @ngInject */ (goToHosting) => goToHosting,
+
+    goBackWithError: /* @ngInject */ ($translate, goBack) => (error) =>
+      goBack(
+        $translate.instant('hosting_dashboard_cdn_order_error', {
+          message: error,
+        }),
+        'danger',
+      ),
+
+    isOptionFree: /* @ngInject */ (serviceOption) =>
+      serviceOption.planCode === 'cdn_free_business',
+
+    serviceName: /* @ngInject */ ($transition$) =>
+      $transition$.params().productId,
+
+    serviceOption: /* @ngInject */ (availableOptions, goBackWithError) =>
+      find(availableOptions, { family: 'cdn' }) ||
+      goBackWithError('No serviceOption found'),
+
+    serviceInfo: /* @ngInject */ (Hosting, serviceName) =>
+      Hosting.getServiceInfos(serviceName),
+
+    cdnProperties: /* @ngInject */ (
+      HostingCdnSharedService,
+      serviceName,
+      goBack,
+    ) => {
+      return HostingCdnSharedService.getCDNProperties(serviceName)
+        .then(({ data: cdn }) => {
+          if (cdn.version === HOSTING_CDN_ORDER_CDN_VERSION_V2) return goBack();
+          return cdn;
```

Comment:
```suggestion
          cdn.version === HOSTING_CDN_ORDER_CDN_VERSION_V2 ? goBack() : cdn
```
Also even though JS allows to have different return types, I do not expect `cdnProperties` to be a `$state.go` promise